src/App.js

```diff
@@ -1,19 +1,161 @@
 import React, { Component } from 'react';
+import Row from './Row';
 
 export default class App extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      data: [],
+      sort: [],
+      checkSearch: false,
+      checkSort: false,
+      checkSortName: false,
+      checkSortAge: false,
+      item: {},
+      searched: []
+    }
+    this.handleClick = this.handleClick.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleNameSort = this.handleNameSort.bind(this);
+    this.handleAgeSort = this.handleAgeSort.bind(this);
+    this.handleClear = this.handleClear.bind(this);
+  }
 
   componentDidMount() {
     // Example of data fetching from server
     fetch('data.json')
       .then((httpResponse) => httpResponse.json())
-      .then((data) => console.log(data));
+      .then((data) => {
+        this.setState({data: data, store: data, item: data[0]})
+      });
+  }
+  handleClick(item) {
+    this.setState({item: item})
+  }
+  handleSearch(){
+    let search = this.state.data.filter((item,index)=>item.name.includes(this.refs.search.value));
+    search.length == 0 ? this.setState({checkSearch: true, searched: [], item: {}}) :
+    this.setState({checkSearch: false, searched: search, item: search[0]})
+   
+  }
+  handleNameSort() {
+    let array = this.state.searched.length == 0 ? this.state.data : !this.state.checkSearch ? this.state.searched : this.state.data;
+    if (this.state.checkSortName) {
+      let sorted = array.sort((item, nextItem) => (item.name.trim() < nextItem.name.trim()) ? -1 : (item.name.trim() > nextItem.name.trim()) ? 1 : 0);
+      this.setState({search: sorted, item: sorted[0], checkSortName: !this.state.checkSortName}) 
+    }
+    else {
+      let sorted = array.sort((item, nextItem) => (item.name.trim() < nextItem.name.trim()) ? 1 : (item.name.trim() > nextItem.name.trim()) ? -1 : 0);
+      this.setState({search: sorted, item: sorted[0], checkSortName: !this.state.checkSortName})
+    }
+  }
+  handleAgeSort(){
+    let array = this.state.searched.length == 0 ? this.state.data : !this.state.checkSearch ? this.state.searched : this.state.data;
```

Comment:
looks like this checks is duplicated in several methods, what about DRY (Do not repeat yourself) practice ?